test/transform_done_test.cpp

```diff
@@ -57,6 +58,24 @@ TEST(TransformDone, Smoke) {
   EXPECT_EQ(count, 1);
 }
 
+TEST(TransformDone, StayDone) {
+  timed_single_thread_context context;
+
+  auto scheduler = context.get_scheduler();
+
+  int count = 0;
+
+  sequence(
+    schedule_after(scheduler, 200ms)
+      | transform_done(
+          []{ return just_done(); }), 
```

Comment:
This seems like an odd way to test `just_done()` since it's being used to transform the 'done' signal into the 'done' signal.

Have you considered doing something like applying `just_done() | materialize() | sync_wait()` and then checking that the result is `std::optional<std::variant<std::tuple<tag_t<set_done>>>>`?

Or maybe do something similar by using `transform_done()` but this time to have it use `just_done()` as the source instead of the fallback?
```
std::optional<int> result =
  just_done()
  | transform_done([] { return just(42); })
  | sync_wait();
assert(result.has_value());
assert(result.value() == 42);
```